### PR TITLE
:tada: show explorers and incoming sdg references in the charts admin

### PR DESCRIPTION
**Pull request by danyx23**
This PR uses the fact that we now have up-to-date information in our DB of which explorers use which charts (by id). The way this information is used is in the chart admin in the references panel.

To complete the picture, this PR also adds a new table into the DB that stores the chart slugs and ids that are referenced from the [sdg-tracker.org](https://sdg-tracker.org) website and shows this in the references section as well.

### CODE REVIEW

File: adminSiteServer/apiRouter.ts [hunk 1]
```diff
@@ -173,39 +207,45 @@ const getReferencesByChartId = async (
     // `grapher/gdp` is a substring of `grapher/gdp-maddison`. We need to be
     // careful not to erroneously match those, which is why we switched to a
     // REGEXP.
```
Comment (sophiamersmann) [inline adminSiteServer/apiRouter.ts:209]: This comment is now misplaced, right?

**Review by sophiamersmann — APPROVED** [review]
Great to have this information in the admin! 🙌🏻 

(I reviewed faithfully but didn't really understand the migration but I'm sure Marcel will)

File: adminSiteClient/EditorReferencesTab.tsx [hunk 2]
```diff
@@ -1,13 +1,139 @@
...
+        </>
+    ) : (
+        <></>
+    )
+    const legacySdgCharts = props.references?.legacySdgCharts?.length ? (
+        <>
+            <React.Fragment>
+                <p>
+                    Legacy SDG referenced charts (slugs of charts only, links to
+                    precise SDG page are not available):
```
Comment (marcelgerber) [inline adminSiteClient/EditorReferencesTab.tsx:104]: I can generate a list of slugs together with the article they appear, shouldn't be too much work.

File: adminSiteClient/EditorReferencesTab.tsx [hunk 3]
```diff
@@ -1,13 +1,139 @@
...
+                </p>
+                <ul className="list-group">
+                    {props.references.legacySdgCharts.map((sdgSlug) => (
+                        <li key={sdgSlug} className="list-group-item">
+                            sdgSlug
```
Comment (marcelgerber) [inline adminSiteClient/EditorReferencesTab.tsx:109]:
```suggestion
                            {sdgSlug}
```

File: db/migration/1685706058408-AddLegacySdgChartReferences.ts [hunk 4]
```diff
@@ -0,0 +1,597 @@
...
+                [slug, slug]
+            )
+            if (rows.length === 0) {
+                console.error(`No chart found for ${slug}`)
+                continue
+            }
+            if (rows.length > 1)
+                console.error(
+                    `Multiple charts found for ${slug} - selecting the first one`
+                )
```
[the remark this answers is not in the document]
Comment (danyx23) [reply db/migration/1685706058408-AddLegacySdgChartReferences.ts:34]: Yeah good point - added the check for `publishedBy is not null` and extracted a function

File: adminSiteClient/EditorReferencesTab.tsx (hunk 2 above)
Comment (danyx23) [reply adminSiteClient/EditorReferencesTab.tsx:104]: I don't think this is needed at this point - we'll migrate away soon anyhow

**Comment by danyx23** [comment]
@marcelgerber could I have your eyes over this once more after I addressed the comments? I rebased on top of master so the broken staging is because it was created with node 16.

**Pull request merged**